### PR TITLE
feat(Alert): add new bq-alert component

### PR DESCRIPTION
**Pull request by Cata1989**
The Alert is a user interface component used to convey important information to the user in a clear and concise manner. It can be used to notify users of success, failure, warning, or any other type of information that needs to be brought to their attention.

Screenshot 2023-11-29 at 14 36 14

https://github.com/Endava/BEEQ/assets/98550729/5bfe625f-f5b1-45a9-931b-ac8e9d70e91d

### CODE REVIEW

File: packages/beeq/src/components/alert/bq-alert.tsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
+import { Component, Element, Event, EventEmitter, h, Host, Listen, Method, Prop, State, Watch } from '@stencil/core';
```
Comment (dgonzalezr) [inline packages/beeq/src/components/alert/bq-alert.tsx:1]: the `Listen` can be removed, is no longer used in the component implementation 😉

File: packages/beeq/src/components/alert/bq-alert.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+ * @part body - The conatiner `<div>` that wraps the alert description content
+ * @part btn-close - The `bq-button` used to close the alert
+ * @part content - The conatiner `<div>` that wraps all the alert content (title, description, footer)
+ * @part footer - The conatiner `<div>` that wraps the alert footer content
+ * @part icon - The `<bq-icon>` element used to render a predefined icon based on the alert type (info, success, warning, error, default)
+ * @part icon-outline - The conatiner `<div>` that wraps the icon element
+ * @part main - The conatiner `<div>` that wraps the alert main content (title, description)
+ * @part svg - The `<svg>` element of the predefined bq-icon component
+ * @part title - The conatiner `<div>` that wraps the alert title content
+ * @part wrapper - The wrapper container `<div>` of the element inside the shadow DOM
```
Comment (dgonzalezr) [inline packages/beeq/src/components/alert/bq-alert.tsx:17]: There is a typo with `container`, it says `conatiner ` in multiple places.

File: packages/beeq/src/components/alert/bq-alert.tsx (hunk 1 above)
Comment (Cata1989) [reply packages/beeq/src/components/alert/bq-alert.tsx:1]: Ohh, yes, I forgot.

File: packages/beeq/src/components/alert/bq-alert.tsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        aria-hidden={!this.open ? 'true' : 'false'}
+        hidden={!this.open ? 'true' : 'false'}
+        role="alert"
+      >
+        <div
+          class={{
+            'bq-alert': true,
+            [`bq-alert__${this.type}`]: true,
+            'is-sticky': this.sticky,
+          }}
```
Comment (dgonzalezr) [inline packages/beeq/src/components/alert/bq-alert.tsx:193]:
```suggestion
          class={{
            [`bq-alert bq-alert__${this.type}`]: true,
            'is-sticky': this.sticky,
          }}
```

File: packages/beeq/src/components/alert/scss/bq-alert.scss [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  }
+}
+
+.bq-alert {
+  @apply relative flex min-w-[var(--bq-alert--min-width)] p-[var(--bq-alert--padding)];
+  @apply rounded-[var(--bq-alert--border-radius)];
+
+  border-width: var(--bq-alert--border-width);
+  border-style: var(--bq-alert--border-style);
+}
```
Comment (dgonzalezr) [inline packages/beeq/src/components/alert/scss/bq-alert.scss:29]:
```suggestion
.bq-alert {
  @apply relative flex min-w-[var(--bq-alert--min-width)] p-[var(--bq-alert--padding)];
  @apply rounded-[var(--bq-alert--border-radius)] border-[length:--bq-alert--border-width];

  border-style: var(--bq-alert--border-style);
}
```

File: packages/beeq/src/components/alert/scss/bq-alert.scss [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  .bq-alert {
+    @apply items-center justify-center rounded-none;
+  }
+}
+
+.bq-alert {
+  @apply relative flex min-w-[var(--bq-alert--min-width)] p-[var(--bq-alert--padding)];
+  @apply rounded-[var(--bq-alert--border-radius)] border-[length:--bq-alert--border-width];
+  border-style: var(--bq-alert--border-style);
+}
```
Comment (dgonzalezr) [inline packages/beeq/src/components/alert/scss/bq-alert.scss:27]: For entrance animation we can add the following:

```suggestion
.bq-alert {
  @include animation-fade-in;

  @apply relative flex min-w-[var(--bq-alert--min-width)] p-[var(--bq-alert--padding)];
  @apply rounded-[var(--bq-alert--border-radius)] border-[length:--bq-alert--border-width];

  border-style: var(--bq-alert--border-style);
}
```

This `animation-fade-in` is a SCSS mixing that can be found in this file:
`packages/beeq/src/global/styles/mixins/_keyframes.scss`

File: packages/beeq/src/components/alert/scss/bq-alert.variables.scss [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+/* -------------------------------------------------------------------------- */
+
+:host {
+  /**
+   * @prop --bq-alert--background: The alert background color
+   * @prop --bq-alert--border-radius: The alert border radius
+   * @prop --bq-alert--content-footer-gap: The alert content and footer gap
+   * @prop --bq-alert--title-body-gap: The alert title and body gap
+
+   * @prop --bq-alert--box-shadow: The alert box shadow
```
Comment (dgonzalezr) [inline packages/beeq/src/components/alert/scss/bq-alert.variables.scss:12]: I think we don't use `box-shadow` in the Alert component 😉

**Review by dgonzalezr — APPROVED** [review]
Well done @Cata1989 🎉

**Pull request merged**